bin/scanpy_wrapper_utils.py

```diff
@@ -0,0 +1,156 @@
+                break
+            V.append(v)
+        return V
+
+    return parse_comma_separated_list
+
+
+class ScanpyArgParser(object):
+    def __init__(self, description=None):
+        self.parser = argparse.ArgumentParser(description=description)
+        self.parser.add_argument('--debug', action='store_true',
+                                 help='Print debug information.')
+        self.events = [{'handler':'_set_logging_level',
+                        'argv':'debug'}]
+
+
+    def add_input_object(self):
+        self.parser.add_argument('-i', '--input-object-file',
+                                 required=True,
+                                 help='Path to anndata or loom file.')
+        self.parser.add_argument('-f', '--input-format',
+                                 choices=['loom', 'anndata', 'auto-detect'],
+                                 default='auto-detect',
+                                 help='Format for input object: loom/anndata/[auto-detect].')
+        self.events.append({'handler':'_detect_io_format',
+                            'argv':['input_format','input_object_file']})
+
+
+    def add_output_object(self):
+        self.parser.add_argument('-o', '--output-object-file',
+                                 required=True,
+                                 help='File name in which to store serialized python object.')
+        self.parser.add_argument('-F', '--output-format',
+                                 choices=['loom', 'anndata', 'auto-detect'],
+                                 default='auto-detect',
+                                 help='Format for output object: loom/anndata/[auto-detect].')
+        self.events.append({'handler':'_detect_io_format',
+                            'argv':['output_format','output_object_file']})
+
+
+    def add_subset_parameters(self):
+        self.parser.add_argument('-s', '--subset-names',
+                                 required=True,
+                                 type=comma_separated_list('subset-names', 'string'),
+                                 default=[],
+                                 help='Parameters to subset on.')
+        self.parser.add_argument('-l', '--low-thresholds',
+                                 type=comma_separated_list('low-thresholds', 'numeric'),
+                                 default=[],
+                                 help='Low cutoffs for the parameters (default is -Inf).')
+        self.parser.add_argument('-j', '--high-thresholds',
+                                 type=comma_separated_list('high-thresholds', 'numeric'),
+                                 default=[],
+                                 help='High cutoffs for the parameters (default is Inf).')
+        self.events.append({'handler':'_check_parameter_range',
+                            'argv':['subset_names','low_thresholds','high_thresholds']})
+
+
+    def _set_logging_level(self, argv):
```

Comment:
what is argv supposed to be here? a class or a logging level? Can you please make it explicit in the argument name (instead of argv)? Thanks!